presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -514,6 +513,48 @@ private static Slice formatDatetime(ISOChronology chronology, Locale locale, lon
         }
     }
 
+    @Description("Returns the current Unix timestamp in seconds (seconds since Unix epoch)")
+    @ScalarFunction
+    @SqlType(StandardTypes.BIGINT)
+    public static long unixTimestamp(ConnectorSession session)
+    {
+        return MILLISECONDS.toSeconds(session.getStartTime());
```

Comment:
`startTime` is the start of the query. All of the date functions use it so that they return consistent results for the duration of the query.